x-pack/filebeat/input/awss3/states.go

```diff
@@ -21,7 +21,7 @@ const awsS3ObjectStatePrefix = "filebeat::aws-s3::state::"
 type states struct {
 	// Completed S3 object states, indexed by state ID.
 	// statesLock must be held to access states.
-	states     map[string]state
+	states     map[string]*state
```

Comment:
Is there a reason this was changed to a pointer? This means the mutex is no longer a reliable lock on changes to the table entries.

Reply:
This reduces memory usage when we perform state removals. Using a pointer here allows us to only allocate the pointer size instead of the struct size. As Go Maps can only grow, this benefits the current memory pressure related issues.

Regarding immutability, as long as the ID is fixed (which seems to be the case per code), there shouldn't be any state issue. But let me know if you see something different here